types/dummy/index.d.ts

```diff
@@ -3,3 +3,25 @@
  * ember-data. They will not be shipped to consumers. Ember-data still relies
  * on some private Ember APIs -- those should be defined here as we encounter them.
  */
+declare module '@ember/debug' {
+  export function assert(msg: string, test: any): boolean;
```

Comment:
should test be `Function | boolean`

Reply:
@igorT I usually rely on the API docs as the source of truth, and they don't indicate that a predicate can be passed in.
https://www.emberjs.com/api/ember/3.4/functions/@ember%2Fdebug/assert

I could see where passing a predicate could be confusing
```ts
assert('Problem with toString', obj.toString);
```
Does this ☝️ mean "the value of `.toString` was falsy", or "the value returned by `obj.toString()` was falsy"?


RE: whether the types should be imported vs. explicitly stated w/in ember-data, `assert` should be provided by the `@types/ember__debug` package. No need for it to be defined in this project.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/549360d30e796ac5635e019cb276e8af9933ff71/types/ember__debug/index.d.ts#L10

Reply:
I'll look into whether I can import this type correctly, it wasn't resolving with the installed ember types but perhaps I just need to add the debug module.

Reply:
@types/ember has @types/ember__debug as a dependency. Should work.